src/main/java/net/dv8tion/jda/api/entities/Role.java

```diff
@@ -373,5 +373,47 @@ default String getIntegrationId()
         {
             return isIntegration() ? Long.toUnsignedString(getIntegrationIdLong()) : null;
         }
+
+        /**
+         * Whether this role can be acquired through a premium subscription purchase.
+         *
+         * @return True, if this is a subscription role
+         *
+         * @see    #getSubscriptionIdLong()
+         */
+        boolean hasSubscriptionListing();
+
+        /**
+         * The subscription listing id for this role.
+         *
+         * @return The listing id, or 0 if this role is not for a subscription listing
+         */
+        long getSubscriptionIdLong();
+
+        /**
+         * The subscription listing id for this role.
+         *
+         * @return The listing id, or null if this role is not for a subscription listing
+         */
+        @Nullable
+        default String getSubscriptionId()
+        {
+            return hasSubscriptionListing() ? Long.toUnsignedString(getSubscriptionIdLong()) : null;
+        }
+
+        /**
+         * Whether this role can be purchased by a user.
+         *
+         * @return True, if this role is purchasable
+         */
```

Comment:
```suggestion
        /**
         * Whether this role has been published for user purchasing. Only {@link #hasSubscriptionListing() premium roles} 
         * can be purchased. However, a premium role must be published before it can be purchased. 
         * Additionally, a premium role can be unpublished after it has been published. Doing so will make it 
         * no longer available for purchase but will not remove the role from users who have already purchased it.
         *
         * @return True, if this role is purchasable
         *
         * @see    #hasSubscriptionListing()
         */
```